### ISSUE TITLE
Globals set by env() not available in commands

### ISSUE THREAD

**Issue by jshumaker-accusoft** [yargs/yargs#568]
Following test code:
```
var argv = require('yargs')
  .env('FIZZ')
  .option('test', {
      global: true
    })
  .command('buzz', 'test demanded global', function (yargs) {
    return yargs.demand(['test']);
  })
  .argv;
console.log(argv);
```

Define a FIZZ_TEST environment variable and then run without the command, you see the expected value for test.
`FIZZ_TEST=value node test.js`

Run with the command, `FIZZ_TEST=value node test.js buzz` and you get an error about test not set, remove the demand() from the builder, you can see that test is undefined for the command.

Config files don't have this issue, the options set from the config does propagate into the command.

Workaround I'm using in the meantime is to run the .env() call again within the command builder.  This does show consistent behavior:

```
var argv = require('yargs')
  .env('FIZZ')
  .option('test', {
      global: true
    })
  .command('buzz', 'test demanded global', function (yargs) {
    return yargs.env('FIZZ').demand(['test']);
  })
  .argv;
console.log(argv);
```

However, it means that across all my commands which share some global options, they have to duplicate the .env() call. It'd be nice if either this wasn't necessary and .env() functioned like .config(), or at least the docs pointed out that anything done by .env() will be reset in commands.

**Comment by nexdrew** [comment]
@jshumaker-accusoft Hi! This seems to be a duplicate of #486. Good news is that we have an open PR to fix this in #553. 😃  Bad news is it might not make it into yargs until 5.x. 😞 Regardless, thanks for reporting, and let us know your thoughts. 👍

**Comment by jshumaker-accusoft** [comment]
Doh thanks, tried to find an existing issue, but failed apparently.
